src/librustc/traits/error_reporting.rs

```diff
@@ -1,3 +1,5 @@
+// ignore-tidy-filelength
```

Comment:
Time to split this file up then or someone should sign up to do it before the PR lands I'd say.

Reply:
I'd like to tackle splitting this file, if that's a blocker - we'll see how it goes :-)

Reply:
(The huge diff numbers made the PR disproportionately more intimidating to review :crying_cat_face:  )